src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+
+    private int size;
+
+    private void isIndexOk(int index) {
+        if (index < 0 || index + 1 > size) {
+            throw new IndexOutOfBoundsException("Index can`t be < 0 and > size, index = " + index
+                    + " size = " + size);
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> oldNode = first;
+        for (int i = 0; i < index; i++) {
+            oldNode = oldNode.next;
+        }
+        return oldNode;
+    }
+
+    private void unLink(Node<T> node) {
+        Node<T> prev = node.previous;
+        Node<T> next = node.next;
+        if (size == 1) {
+            first = null;
+            last = null;
+        }
+        if (next == null && prev == null) {
+            size = 1;
+            return;
+        }
+
+        if (next == null) {
+            prev.next = null;
+            last = prev;
+        } else {
+
+            if (prev == null) {
+                next.previous = null;
+                first = next;
+            } else {
+                prev.next = next;
+                next.previous = prev;
+            }
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public void add(T element) {
+        Node<T> newNode = new Node<>(element);
+        if (first == null) {
+            newNode.next = null;
+            newNode.previous = null;
```

Comment:
you created newNode on line 55 with element only, so next and previous are already null, no need to set it again
```suggestion
```